File: php/src/Structural/Adapter/TeaBall.php
<?

#//TeaBall.java - the adapter

require_once 'TeaBag.php';


class TeaBall extends TeaBag
{

	public $looseLeafTea;

	function __construct($looseLeafTea)
	{
		$this->looseLeafTea = $looseLeafTea;
		$this->teaBagIsSteeped = $looseLeafTea->teaIsSteeped;
	}
		
	function steepTeaInCup ()
	{
		$this->looseLeafTea->steepTea();
		$this->teaBagIsSteeped=1;
	}

}

File: csharp/src/Structural/Adapter/TeaBall.cs
namespace Structural.Adapter {


//TeaBall.java - the adapter

public class TeaBall : TeaBag {  
   LooseLeafTea looseLeafTea;
   
   public TeaBall(LooseLeafTea looseLeafTeaIn) {
       looseLeafTea = looseLeafTeaIn;
       teaBagIsSteeped = looseLeafTea.teaIsSteeped;
   }
    
   public new void steepTeaInCup() {
       looseLeafTea.steepTea();
       teaBagIsSteeped = true;
   }
}

}
